Lib/multiprocessing/queues.py

```diff
@@ -313,30 +313,47 @@ class SimpleQueue(object):
     def __init__(self, *, ctx):
         self._reader, self._writer = connection.Pipe(duplex=False)
         self._rlock = ctx.Lock()
-        self._poll = self._reader.poll
         if sys.platform == 'win32':
             self._wlock = None
         else:
             self._wlock = ctx.Lock()
 
+    def _check_closed(self):
+        if self._reader is None:
+            raise ValueError("operation on closed queue")
+
+    def close(self):
+        try:
+            reader = self._reader
+            if reader is not None:
+                self._reader = None
+                reader.close()
+        finally:
+            writer = self._writer
+            if writer is not None:
+                self._writer = None
+                writer.close()
+
     def empty(self):
-        return not self._poll()
+        self._check_closed()
+        return not self._reader.poll()
 
     def __getstate__(self):
         context.assert_spawning(self)
         return (self._reader, self._writer, self._rlock, self._wlock)
 
     def __setstate__(self, state):
         (self._reader, self._writer, self._rlock, self._wlock) = state
-        self._poll = self._reader.poll
 
     def get(self):
+        self._check_closed()
```

Comment:
SimpleQueue is supposed to be thread-safe? (I don't know the answer, it's a real question!) If we start to use the read lock, we also may have to use the write lock??